test/unit/throttle_enumerator_test.rb

```diff
@@ -110,6 +110,24 @@ def each_iteration(record, _params)
       assert_equal 0, IterationThrottleJob.on_complete_called
     end
 
+    test "does not evaluate enumerator when throttled" do
+      class FlunkOnIterationThrottleJob < ActiveJob::Base
```

Comment:
Looks good except we should avoid be declaring constants (such as classes) in test bodies.

Can we move this to the test class body instead?

Reply:
Fixed